### ISSUE TITLE
SIGSEGV on introspection POST with username

### ISSUE THREAD

**Issue by mimi89999** [authelia/authelia#6485]
### Version

v4.37.5

### Deployment Method

Bare-metal

### Reverse Proxy

NGINX

### Description

SIGSEGV on introspection POST with username

### Reproduction

```bash
curl -X POST -d "token=authelia_at_aaa" "https://myapp@auth.lebihan.pl/api/oidc/introspection"
```

### Expectations

Should not crash

### Configuration (Authelia)

```yaml
identity_providers:
  oidc:
    hmac_secret: SECRET
    issuer_private_key: |
      -----BEGIN PRIVATE KEY-----
      SECRET
      -----END PRIVATE KEY-----
      clients:
        id: myapp
        description: My Application
        public: True
        authorization_policy: one_factor
        scopes:
          - openid
          - groups
          - email
          - profile
        redirect_uris:
        - com.fsck.k9://oauth2redirect
        - com.fsck.k9.debug://oauth2redirect
```

### Build Information

```shell
Last Tag: v4.37.5
State: tagged clean
Branch: v4.37.5
Commit: 566a0d7fc71b450123ad33d350cd3890d311da82
Build Number: 17068
Build OS: linux
Build Arch: arm64
Build Date: Wed, 21 Dec 2022 19:54:54 +1100
Extra:
```

### Logs (Authelia)

```shell
Jan 01 00:41:06 kokomi systemd[1]: Started authelia.service - Authelia authentication and authorization server.
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=warning msg="Configuration: access control: no rules have been specified so the 'default_policy' of 'one_factor' is going to be applied to all requests"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=info msg="Authelia v4.37.5 is starting"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=info msg="Log severity set to debug"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=debug msg="Registering client myapp with policy one_factor (one_factor)"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=info msg="Storage schema is being checked for updates"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=info msg="Storage schema is already up to date"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=debug msg="The NTP startup check was skipped due to there being no configured 2FA access control rules"
Jan 01 00:41:06 kokomi authelia[8252]: time="2024-01-01T00:41:06+01:00" level=info msg="Initializing server for non-TLS connections on '[::]:9091' path '/'"
Jan 01 00:41:14 kokomi authelia[8252]: panic: runtime error: invalid memory address or nil pointer dereference
Jan 01 00:41:14 kokomi authelia[8252]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5595159ee0]
Jan 01 00:41:14 kokomi authelia[8252]: goroutine 28 [running]:
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/configuration/schema.(*PasswordDigest).Encode(0x4000a66190?)
Jan 01 00:41:14 kokomi authelia[8252]:         <autogenerated>:1 +0x20
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/oidc.(*Client).GetHashedSecret(0x4000134240?)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/oidc/client.go:83 +0x30
Jan 01 00:41:14 kokomi authelia[8252]: github.com/ory/fosite.(*Fosite).checkClientSecret(0x400026bcc0, {0x559697c248, 0x400046d560}, {0x5596981650?, 0x4000e39200}, {0x55973b3128, 0x0, 0x0})
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/ory/fosite@v0.44.0/client_authentication.go:240 +0x6c
Jan 01 00:41:14 kokomi authelia[8252]: github.com/ory/fosite.(*Fosite).NewIntrospectionRequest(0x400026bcc0, {0x559697c478?, 0x4000158000?}, 0x4000e38000, {0x559697d560, 0x400046d0b0})
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/ory/fosite@v0.44.0/introspection_request_handler.go:142 +0xcec
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/handlers.OAuthIntrospectionPOST(0x4000158000, {0x5596979470, 0x400046d080}, 0x40001ba070?)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/handlers/handler_oauth_introspection.go:23 +0x16c
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.NewHTTPToAutheliaHandlerAdaptor.func1(0x4000158000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/http_to_authelia_handler_adaptor.go:113 +0x2d0
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.(*BridgeBuilder).Build.func1.1(0x4001032330?)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/bridge.go:54 +0xb4
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.SecurityHeadersNoStore.func1(0x4001032000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/headers.go:35 +0xa0
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.SecurityHeadersCSPNone.func1(0x4001032000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/headers.go:25 +0x6c
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.SecurityHeaders.func1(0x4001032000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/headers.go:16 +0x130
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.(*CORSPolicy).Middleware.func1(0x40001e4d80?)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/cors.go:216 +0x44
Jan 01 00:41:14 kokomi authelia[8252]: github.com/fasthttp/router.(*Router).Handler(0x40001e0000, 0x4001032000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/fasthttp/router@v1.4.14/router.go:414 +0x3f8
Jan 01 00:41:14 kokomi authelia[8252]: github.com/authelia/authelia/v4/internal/middlewares.LogRequest.func1(0x4001032000)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/authelia/authelia/v4/internal/middlewares/log_request.go:14 +0x94
Jan 01 00:41:14 kokomi authelia[8252]: github.com/valyala/fasthttp.(*Server).serveConn(0x4000134480, {0x5596981a70?, 0x400031e010})
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/valyala/fasthttp@v1.43.0/server.go:2338 +0xdf4
Jan 01 00:41:14 kokomi authelia[8252]: github.com/valyala/fasthttp.(*workerPool).workerFunc(0x4000a57a40, 0x4000e08020)
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/valyala/fasthttp@v1.43.0/workerpool.go:224 +0x70
Jan 01 00:41:14 kokomi authelia[8252]: github.com/valyala/fasthttp.(*workerPool).getCh.func1()
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/valyala/fasthttp@v1.43.0/workerpool.go:196 +0x38
Jan 01 00:41:14 kokomi authelia[8252]: created by github.com/valyala/fasthttp.(*workerPool).getCh
Jan 01 00:41:14 kokomi authelia[8252]:         github.com/valyala/fasthttp@v1.43.0/workerpool.go:195 +0x220
Jan 01 00:41:14 kokomi systemd[1]: authelia.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jan 01 00:41:14 kokomi systemd[1]: authelia.service: Failed with result 'exit-code'.
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
You must submit the full logs and config, especially the OIDC config.

**Comment by james-d-elliott** [comment]
Fixed by 8e8d07ab75d14e7c98f045e242a96b84d5e4e36e
